job-service-db/README_functions.md

```diff
@@ -8,24 +8,63 @@ differ)
 
 ## Adding a function
 
-When adding an **internal function**, we would need to forward declare it in the migration script. Ex:  
+### External function:
+- Create a new file with the function's name preceded with `R__` within the `procedures` folder  
+### Internal function:  
+- Add `internal__` as prefix for the name of the function 
+- Add `R__internal__` as prefix for the function's file.
+- Forward declare the function in the migration script. 
+- Ensure that we have `/* Forward Declaration */` in the body of the forward declared function.
+
+If working on a `plpgsql` function, declare the body as:  
+`AS 'BEGIN /* Forward Declaration */ END';`  
+
+Ex:  
+
+    DO $$
+    BEGIN
+    CREATE FUNCTION internal_get_task_status(
```

Comment:
Add some indenting here.  Indent all the rows between `BEGIN` and `EXCEPTION`.

Reply:
Done